src/Spanner/Session/CacheSessionPool.php

```diff
@@ -257,6 +268,11 @@ public function acquire($context = SessionPoolInterface::CONTEXT_READ)
             $session = $this->waitForNextAvailableSession();
         }
 
+        if ($this->deleteQueue) {
+            $this->deleteSessions($this->deleteQueue);
```

Comment:
`manageSessionsToDelete` is called when a lock is acquired, and I wanted to avoid any network requests being triggered during that state.